### DISCUSSION TITLE
AutoScroll to the End after each Log entry

### DISCUSSION THREAD

**Discussion by KayakFisher205** [augustoproiete/serilog-sinks-richtextbox#22] [Q&A]
I tested the Samples (WinFormsHostNet50Sample, WpfNet50Sample, and WpfNet462Sample) and when the Log reached the bottom of the form, it does not scroll. When I use Serilog to log to the Console, it does. I'm using the Hybrid WPF/Winforms Example (WinFormsHostNet50Sample), so I can't use a trigger or Event (that I know of). If its possible to enable ScrollToEnd() somehow, or give the option to reverse the log (richtextbox.Text="new Log data" + richtextbox.Text), that would put the newest log on the top.

I use the program to monitor alot of processes so it would be helpful to see the most recent output without manually scrolling down.

**Comment by ghost** [comment]
Hi i have posted @stackoverflow a solution for wpf richtextbox autoscroll.
Just replace in stackoverflow code "yourRichtTextBox"  with "_richTextBox" from the WpfNet50Sample code,

[https://stackoverflow.com/a/69489605/5342618](https://stackoverflow.com/a/69489605/5342618])

**Reply by KayakFisher205** [reply]
Thanks for the help. I usually do really good finding things, especially on StackOverFlow, but I must have missed it. Works like a charm!

**Comment by KayakFisher205** [comment]
I noticed when the form does not have focus or I lock my desktop (which the form might not have focus), the autoscroll isn't working. What would the code be to add to autoscroll to the bottom when the form got focus? I have not used WPF at all.

**Reply by ghost** [reply]
The autoscroll works for me without form focus and also when the desktop is locked. If you like you can upload a example.

**Reply by ghost** [reply]
I have added a new feature thats maybe something for you. I find out that the LogView use a lot of RAM with much Text. I have write some code to limit the amount of text in the RichTextBox.  

[https://pastebin.com/T8sqD0Tr](https://pastebin.com/T8sqD0Tr)

**Reply by KayakFisher205** [reply]
I thought it was locking the screen and/or losing focus, but I have been unable to recreate the issue although I had seen it twice before I set the message. I will add the new code and give it a run. Thanks for the Update!

**Reply by KayakFisher205** [reply]
Since the way I an using this control is via a global variable, I put the code in my program.cs and I felt I needed to add the line
 wpfRichTextBox.TextChanged += new TextChangedEventHandler(RichTextBox_TextChanged);
since the control is not created in XAML.

I started my code and it is running around 126 Mb of RAM. I'll watch it over the course of the week and see if there is an change.

Thanks again for all the hlep.

**Reply by ghost** [reply]
Did you notice that autoscroll turns off automatically when you scroll manually? It will only be activated again when you scroll down to the bottom. You can disable that in the code if you want.

**Reply by KayakFisher205** [reply]
Can you let me know what its needs to keep Autoscroll on? When I remote desktop to the Server running the code, it seems I have to manually scroll down. Possibly the RD is triggering that.

**Reply by ghost** [reply]
> Can you let me know what its needs to keep Autoscroll on? When I remote desktop to the Server running the code, it seems I have to manually scroll down. Possibly the RD is triggering that.

**remove the following code:**

// User scroll event : set or unset autoscroll mode
if (e.Source as ScrollViewer != null && e.ExtentHeightChange == 0)
{ // Content unchanged : user scroll event
if ((e.Source as ScrollViewer).VerticalOffset == (e.Source as ScrollViewer).ScrollableHeight)
{ // Scroll bar is in bottom
// Set autoscroll mode
AutoScroll = true;
}
else
{ // Scroll bar isn't in bottom
// Unset autoscroll mode
AutoScroll = false;
}
}
